src/utils/url/getAbsoluteUrl.ts

```diff
@@ -2,8 +2,8 @@
  * Returns an absolute URL based on the given path.
  */
 export function getAbsoluteUrl(path: string, baseUrl?: string): string {
-  // Ignore absolute URLs.
-  if (!path.startsWith('/')) {
+  // Ignore url with any protocol or starts with *、:
+  if (/^[a-z][a-z0-9+.-]*:/.test(path) || /^[:*]/.test(path)) {
```

Comment:
Do you think we can achieve the same result by including the `://` in the regular expression?

```suggestion
  if (/^[a-z][a-z0-9+.-]*:\/\//.test(path)) {
```

Reply:
@kettanaito in current test case, the result have no difference. but pattern url like `:api/user`, `*` failed

Reply:
This expression seems to cover the use cases we need:

```js
/^\w+:\/\//
```

Could you please confirm that?

The issue with the current approach is the permissive `*` token, which allows _zero_ or more matches. It's the "zero" part that we should replace with `+` (one or more matches). For the URL to be absolute it must have `something://`.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler#permitted_schemes
custom protocol allow `web+`, so numbers are possible

Reply:
https://github.com/axios/axios/blob/master/lib/helpers/isAbsoluteURL.js
axios regexp allow numbers